src/main/java/uk/gov/pay/adminusers/resources/ServiceResource.java

```diff
@@ -227,30 +229,21 @@ public Response removeUserFromService(@PathParam("serviceExternalId") String ser
 
     @Path("/{serviceExternalId}/stripe-agreement")
     @POST
-    @Produces(APPLICATION_JSON)
     @Consumes(APPLICATION_JSON)
-    public Response createStripeAgreement(@PathParam("serviceExternalId") String serviceExternalId, JsonNode payload) {
-        LOGGER.info("Create stripe agreement POST request - [ {} ]", payload);
-
-        Optional<Errors> maybeErrors = stripeAgreementRequestValidator.validateCreateRequest(payload);
-        if (maybeErrors.isPresent()) {
-            return Response.status(BAD_REQUEST).entity(maybeErrors.get()).build();
-        }
-
-        Optional<ServiceEntity> maybeService = serviceDao.findByExternalId(serviceExternalId);
-        if (!maybeService.isPresent()) {
-            return Response.status(NOT_FOUND).build();
-        }
+    public Response createStripeAgreement(@PathParam("serviceExternalId") String serviceExternalId,
+                                          @NotNull @Valid StripeAgreementRequest stripeAgreementRequest) throws UnknownHostException {
+        LOGGER.info("Create stripe agreement POST request - [ {} ]", stripeAgreementRequest.toString());
 
-        if (stripeAgreementService.findStripeAgreementByServiceId(maybeService.get().getId()).isPresent()) {
-            return Response.status(CONFLICT)
-                    .entity(Errors.from("Stripe agreement information is already stored for this service"))
-                    .build();
+        ServiceEntity service = serviceDao.findByExternalId(serviceExternalId)
+                .orElseThrow( () -> new WebApplicationException(Status.NOT_FOUND));
+        
+        if (stripeAgreementService.findStripeAgreementByServiceId(service.getId()).isPresent()) {
+            throw new WebApplicationException("Stripe agreement information is already stored for this service",
+                    Status.CONFLICT);
         }
 
-        stripeAgreementService.doCreate(
-                maybeService.get().getId(),
-                payload.get(FIELD_IP_ADDRESS).asText(),
+        stripeAgreementService.doCreate(service.getId(),
+                InetAddress.getByName(stripeAgreementRequest.getIpAddress()),
                 LocalDateTime.now(ZoneOffset.UTC));
```

Comment:
Was the question about LocalDateTime resolved?

Reply:
No, I think that will need changing, but rather than confuse commits and PRs I thought that could be done afterwards to keep this refactoring clean.